trans/transducer.py

```diff
@@ -539,6 +539,99 @@ def continue_decoding():
         return Output(action_history, self.decode_encoded_output(input_, action_history),
                       log_p, None)
 
+    def roll_in(self, sample: utils.Sample, rollin: int) -> None:
```

Comment:
The implementation is for a single sample at the moment, the decoder steps could probably be batched (and sampling / exploration as loop). This would likely increase speed.